File: ios/Looks/LooksCamera/LooksCameraManager.m
//
//  LooksCameraManager.m
//  Looks
//
//  Created by Dileepa Chandima on 2022-12-31.
//

#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(LooksCameraManager, RCTViewManager)

RCT_EXTERN_METHOD(takePhoto)
RCT_EXPORT_VIEW_PROPERTY(onResultImageExported, RCTBubblingEventBlock)

@end

File: ios/Looks/LooksCamera/LooksCameraManager.swift
//
//  LooksCameraManager.swift
//  Looks
//
//  Created by Dileepa Chandima on 2022-12-31.
//

import Foundation
import UIKit

@objc(LooksCameraManager)
class LooksCameraManager: RCTViewManager {
  
  private var looksCamera = LooksCamera()
  
  override func view() -> UIView {
    return looksCamera;
  }
  
  @objc func takePhoto(){
    looksCamera.takePhoto()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}

